### ISSUE TITLE
Delete old addon subfolders only after unpacking was ok

### ISSUE THREAD

**Issue by Kiraan75** [grrttedwards/wow-addon-updater#80]
If, for some reason, unpacking does not work properly (e.g. "unexpected error unzipping xxx"), the old addon subfolder(s) should persist.

**Comment by grrttedwards** [comment]
Hey welcome back! :)

From my understanding, if the extract fails, since it’s just an overwrite,
that it will leave any of the existing files in-place. But the problem
still exists that it could get halfway through an extract and fail, I’m not
really sure what happens then, since I haven’t really read the ZipFile
extract code.

Could you list out a simple test case that I can reproduce?

For GitHub repos or the subfolder extraction branch, it’s a little more
complicated so I’ll take a look into that.

**Comment by Kiraan75** [comment]
Hi Garrett,

On my side, I had a problem with the CTmod addon. I used the pipe ( | ) delimiter to extract only a specific subfolder and that led to "unexpected error unzipping xxx". I'm not sure if I did something wrong with the formal specification of the URL with subfolder, I'll test it with another .zip.

Nonetheless, that leaves me with deleted addon folders. I think it starts from line 135 on, in addon_manager.py (I'm just a beginner in learning Python, so that's my guess):

                 if isdir(destination_dir):
                    shutil.rmtree(destination_dir)
                shutil.copytree(src=temp_source_dir, dst=destination_dir)

Since all exceptions in update_addon( ) set latest_version = AddonManager._UNAVAILABLE, I think there should be a check and if true, don't delete the destination_dir.

**Comment by grrttedwards** [comment]
Ooh. Yeah the subfolders is what I was hoping was _not_ the way you were using it. 😛 

In order for that to happen, then the subfolder you specified probably does not exist. Let me take a look.

To be specific, I think what happened was:

1. You have an addon like `https://www.curseforge.com/wow/addons/ctmod|CT_BarMod`
2. The addon is extracted to a temp directory successfully
3. The existing installed addon directory is removed
4. The code tries to copy your subfolder from the temp directory into the destination directory and fails for some reason

**Comment by grrttedwards** [comment]
Raised #81 which fixes deleting the existing directory when it shouldn't. But you've found another bug involving Curse archives and subfolders!

I'll create another issue for that.

**Comment by grrttedwards** [comment]
Released [v1.2.2](https://github.com/grrttedwards/wow-addon-updater/releases/tag/v1.2.2) with this fix! Thanks for the report. I'll try to get Curse subfolders working soon.

**Comment by Kiraan75** [comment]
Nice solution to check 
                if not isdir(norm_src_dir):
                    raise KeyError()
I was thinking too complicated (as always...)  Thanks, Sensei :-)
